src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1594,8 +1613,12 @@ def _update_addons(instance, addons, enable, workspace_resource_id=None, no_wait
                     raise CLIError('The monitoring addon is already enabled for this managed cluster.\n'
                                    'To change monitoring configuration, run "az aks disable-addons -monitoring"'
                                    'before enabling it again.')
-                elif workspace_resource_id:
-                    addon_profile.config = {'logAnalyticsWorkspaceResourceID': workspace_resource_id}
+                if not workspace_resource_id:
+                    workspace_resource_id = _ensure_default_log_analytics_workspace_for_monitoring(
+                        cmd,
+                        subscription_id,
```

Comment:
Looks like this subscription id should be selected from the workspace.

Reply:
Talked with Ganga, since this is only being used for the default workspace which is always in the subscription the cluster is in. We don't need to update the id.